src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_params.py

```diff
@@ -34,3 +36,15 @@ def load_arguments(self, _):
         cmp.argument('top', options_list=['--top', '-t'], type=int, help='Maximum number of items to return. Value range: 1-1000.')
         cmp.argument('start_date', options_list=['--start-date', '-s'], type=get_datetime_type(), help='Start date (YYYY-MM-DD in UTC). If specified, also requires --end-date.')
         cmp.argument('end_date', options_list=['--end-date', '-e'], type=get_datetime_type(), help='End date (YYYY-MM-DD in UTC). If specified, also requires --start-date.')
+
+    with self.argument_context('consumption budget') as cb:
+        cb.argument('resource_group_name', options_list='--resource-group-name', type=str, help='Resource group of the budget.')
+        cb.argument('budget_name', options_list='--budget-name', type=str, help='Name of a budget.')
+        cb.argument('category', options_list=['--category', '-c'], type=str, help='Category of the budget can be cost or usage.')
+        cb.argument('amount', options_list=['--amount', '-a'], type=get_decimal_type(), help='Amount of a budget.')
+        cb.argument('time_grain', options_list='--time-grain', type=str, help='Time grain of the budget can be monthly, quarterly, or annually.')
+        cb.argument('start_date', options_list=['--start-date', '-s'], type=get_datetime_type(), help='Start date (YYYY-MM-DD in UTC) of time period of a budget.')
+        cb.argument('end_date', options_list=['--end-date', '-e'], type=get_datetime_type(), help='End date (YYYY-MM-DD in UTC) of time period of a budget.')
+        cb.argument('resource_groups', options_list='--resource-groups', nargs='+', help='Space-separated list of resource groups to filter on.')
```

Comment:
Options list not required. These filter type arguments should have `arg_group='Filter'` so they stand out in help.  Right now, you'll have 

```
--resource-group -g  The "resource group"
--resource-groups     Resources groups to filter on
```

You might also consider making the property name something like "--filter-groups" or something. Right now, `--resource-group` and `--resource-groups` would be easy to confuse.